tests/test_CompensatedGaussianFlux.py

```diff
@@ -0,0 +1,330 @@
+                    exposure.image.array += dc_offset
+                    task.run(catalog, exposure)
+
+                    filter_flux = catalog[f"base_CompensatedGaussianFlux_{width}_instFlux"]
+                    filter_err = catalog[f"base_CompensatedGaussianFlux_{width}_instFluxErr"]
+                    truth_flux = catalog["truth_instFlux"]
+
+                    if width == self.psf_size:
+                        # When the filter matches the PSF, we should get close to the true flux.
+                        tol = np.sqrt((filter_err/filter_flux)**2. + 0.02**2.)
+                        self.assertFloatsAlmostEqual(filter_flux, truth_flux, rtol=tol)
+                    elif width > self.psf_size:
+                        # When the filter is larger than the PSF, the filter flux will be
+                        # greater than the truth flux.
+                        np.testing.assert_array_less(truth_flux, filter_flux)
+
+                    if dc_offset == 0.0:
+                        # Use the no-offset run as a comparison for offset runs.
+                        flux_0 = filter_flux
+                    else:
+                        # Note: this tolerance is determined empirically, but this is
+                        # larger than preferable.
+                        self.assertFloatsAlmostEqual(filter_flux, flux_0, rtol=5e-3)
+
+                    # The ratio of the filter flux to the truth flux should be consistent.
+                    # I'm not sure how to scale this with the error, so this is a loose
+                    # tolerance now.
+                    ratio = filter_flux / truth_flux
+                    self.assertLess(np.std(ratio), 0.04)
+
+    def testMonteCarlo(self):
+        """Test an ideal simulation, with no noise.
+
+        Demonstrate that:
+
+        - We get exactly the right answer, and
+        - The reported uncertainty agrees with a Monte Carlo test of the noise.
+        """
+        nSamples = 500
+
+        for width in (3, 5):
+            for t in (1.5, 2.0):
+                config = lsst.meas.base.SingleFrameCompensatedGaussianFluxConfig()
+                config.kernel_widths = [width]
+                config.t = t
+
+                algorithm, schema = self.makeAlgorithm(config=config)
+
+                # Make a noiseless catalog.
+                exposure, catalog = self.dataset_single.realize(1E-8, schema, randomSeed=1)
+
+                # Only use the high-flux source for the error tests.
+                record = catalog[0]
+                algorithm.measure(record, exposure)
+                inst_flux = record[f"base_CompensatedGaussianFlux_{width}_instFlux"]
+
+                for noise in (0.001, 0.01, 0.1):
+                    fluxes = np.zeros(nSamples)
+                    errs = np.zeros_like(fluxes)
+
+                    for repeat in range(nSamples):
+                        # By using ``repeat`` to seed the RNG, we get results which
```

Comment:
This is basically my problem with Monte Carlo unit tests. What you have here is fine, if possibly not strictly Monte Carlo. It is a good compromise and test behavior sufficiently

Reply:
I have just copied this format from all the other tests in meas_base.